### PR TITLE
New Bastion Fort starting scenario

### PR DESCRIPTION
**Pull request by JonathanLochridge**
#### Summary
Content "New Bastion Fort starting scenario"

#### Purpose of change
I thought it would be cool to be able to start in a bastion fort and have to escape out the single exit or brake a leg going over the wall.

#### Describe the solution
Added a New Scenario which allows starting at the Bastion Fort.
Also buffed the Forts to spawn 0-2 times per overmap square to ensure one spawns so it will work on default world settings.

#### Describe alternatives you've considered
Not adding it.
Or making the spawn rate slightly lower or higher.

#### Testing
I ran all the edited files through the Linter.
I booted it up and made fresh worlds and characters till I consistently got ones that spawned properly.

#### Additional context
The next step would be making the Reenactor the default profession for this start. 
I think paring down the allowed professions would be a good thing to do.
I think in it's current form it is worth merging. But I am planning to do some refinement to make it a bit better.
I also ended up with more commits than I would prefer for what got done. I can't remember how to squash commits well. But I wouldn't mind some of these being squashed together.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
> Also buffed the Forts to spawn 0-2 times per overmap square

That's not the way it works with `UNIQUE` flag. This flag means, that with your change now there's a chance of 20 out of 100 (i.e. 20%) of spawning a single location on any overmap. In other words, this flag converts _amount_ of spawned location into a _probability_ of spawning.

**Comment by JonathanLochridge** [comment]
> > Also buffed the Forts to spawn 0-2 times per overmap square
> 
> That's not the way it works with `UNIQUE` flag. This flag means, that with your change now there's a chance of 20 out of 100 (i.e. 20%) of spawning a single location on any overmap. In other words, this flag converts _amount_ of spawned location into a _probability_ of spawning.

I had temporarily removed the Unique Flag to try to get it to spawn. But I placed it back. With the current version 20% chance of spawning is the expected rate.
Apparently the message for being unable to spawn actually occurs after checking 100 overmaps not 3.

File: data/json/overmap/overmap_special/specials.json [hunk 1]
```diff
@@ -7491,10 +7491,10 @@
       { "point": [ 3, 2, 1 ], "overmap": "fort_4c_1_north" }
     ],
     "locations": [ "wilderness" ],
-    "city_distance": [ 10, 100 ],
+    "city_distance": [ 0, 100 ],
     "city_sizes": [ 0, 12 ],
-    "occurrences": [ 10, 100 ],
-    "flags": [ "CLASSIC", "UNIQUE" ]
+    "occurrences": [ 20, 100 ],
+    "flags": [ "CLASSIC","UNIQUE" ]
```
Comment (Maleclypse) [inline data/json/overmap/overmap_special/specials.json:7497]:
```suggestion
    "flags": [ "CLASSIC", "UNIQUE" ]
```

File: data/json/overmap/overmap_special/specials.json [hunk 2]
```diff
@@ -7491,10 +7491,10 @@
       { "point": [ 3, 2, 1 ], "overmap": "fort_4c_1_north" }
     ],
     "locations": [ "wilderness" ],
-    "city_distance": [ 10, 100 ],
+    "city_distance": [ 0, 100 ],
```
Comment (Maleclypse) [inline data/json/overmap/overmap_special/specials.json:7494]:
```suggestion
    "city_distance": [ 0, -1 ],
```

**Comment by JonathanLochridge** [comment]
Hmm, I guess one of the minor edits must have messed up the Lint. Probably should have reran it. after my final adjustments but I forgot. I can do that now then?

File: data/json/overmap/overmap_special/specials.json (hunk 2 above)
Comment (JonathanLochridge) [reply data/json/overmap/overmap_special/specials.json:7494]: What does having a max city distance of -1 do? Does that give it no max?

**Comment by JonathanLochridge** [comment]
So the change to make the make distance -1 seems to have triggered an error in the build process?
@Maleclypse

**Comment by Maleclypse** [comment]
> So the change to make the make distance -1 seems to have triggered an error in the build process? @Maleclypse

I was going to check the error but you've reverted it. No stress.  If this works now leave it be.

**Comment by JonathanLochridge** [comment]
> > So the change to make the make distance -1 seems to have triggered an error in the build process? @Maleclypse
> 
> I was going to check the error but you've reverted it. No stress. If this works now leave it be.

Oh, sorry I should have waited I guess? I think it should be in the history so you might be able to check?
But, It specifically pointed out that line in the build process. Not sure why? I will wait and see if there is still an issue after it runs again.

**Pull request merged**